sky/provision/aws/instance.py

```diff
@@ -81,9 +82,27 @@ def terminate_instances(region: str,
     instances = _filter_instances(ec2, filters, included_instances,
                                   excluded_instances)
     instances.terminate()
+    # Wait for all instances to be terminated, since we need to delete the
+    # Security Group dependent on them.
+    for instance in instances:
+        instance.wait_until_terminated()
```

Comment:
Actually, this reminds me:

- Does this PR change all AWS clusters' behavior to be "create its own SG + delete its own SG"?
- Can we make it so that we do the per-cluster SG whenever exposed ports are requested?
- Given the above, is it possible to only pay for this wait whenever there's SG to clean up?